server/src/main/java/com/ncc/neon/controllers/BetterController.java

```diff
@@ -129,7 +136,29 @@
         return res;
     }
 
-    @GetMapping(path="preprocess")
+    @GetMapping(path="irsearch")
+    Mono<Object> irsearch(@RequestParam("query") String query, @RequestParam("module") String module)  {
+        // synchronous service
+        return moduleService.buildNlpModuleClient(module)
+                .flatMap(nlpModule -> {
+                    IRNlpModule irModule = (IRNlpModule) nlpModule;
+                    return irModule.searchIR(query);
+                });
+        // static cast from nlpModule -> IR
+        // return Mono String[]
+    }
+
+    @GetMapping(path = "getdoc")
+    Mono<?> getdoc(@RequestParam("docId") String id) {
+        return betterFileService.getById(id);
+    }
+
+    @GetMapping(path = "getirdocs")
+    ArrayList<Docfile> getirdocs(@RequestParam("docIds") String[] ids) throws IOException {
```

Comment:
Is it important that this returns an `ArrayList` vs. a `List`? If not, I think it's better style to make it a `List`